src/fides/api/models/privacy_preference.py

```diff
@@ -162,22 +162,30 @@ def provided_identity_id(cls) -> Column:
     feature = Column(
         Integer, index=True
     )  # When saving privacy preferences with respect to a TCF feature directly
-    purpose = Column(
+    purpose_consent = Column(
         Integer, index=True
-    )  # When saving privacy preferences with respect to a TCF purpose directly
+    )  # When saving privacy preferences with respect to a TCF purpose with a consent legal basis
+    purpose_legitimate_interests = Column(
+        Integer, index=True
+    )  # When saving privacy preferences with respect to a TCF purpose with a legitimate interests legal basis
     special_feature = Column(
         Integer, index=True
     )  # When saving privacy preferences with respect to a TCF special feature directly
     special_purpose = Column(
         Integer, index=True
     )  # When saving privacy preferences with respect to a TCF special purpose directly
-    vendor = Column(
+    vendor_consent = Column(
         String, index=True
-    )  # When saving privacy preferences with respect to a vendor directly. Vendors can apply to multiple systems.
-    system = Column(
+    )  # When saving privacy preferences with respect to a vendor with a legal basis of consent
+    vendor_legitimate_interests = Column(
         String, index=True
-    )  # When saving privacy preferences with respect to a system id directly, in the case where the vendor is unknown
-
+    )  # When saving privacy preferences with respect to a vendor with a legal basis of legitimate interests
+    system_consent = Column(
+        String, index=True
+    )  # When saving privacy preferences with respect to a system id with consent legal basis, in the case where the vendor is unknown
+    system_legitimate_interests = Column(
+        String, index=True
+    )  # When saving privacy preferences with respect to a system id with legitimate interests legal basis, in the case where the vendor is unknown
```

Comment:
We decided to store this flat, making the table wider, to keep all of the sections working the same.  Here, I am adding new columns to both the PrivacyPreferenceHistory(stores every preference saved) and the ServedNoticeHistory table (stores every consent item served)